src/components/molecules/CalendarBodyDates/CalendarBodyDates.tsx

```diff
@@ -0,0 +1,64 @@
+import { Calendar, DateObj, GetDatePropsOptions } from 'dayzed'
+import { Box, Button, Text, Wrap, WrapItem } from '@chakra-ui/react'
+import { FC } from 'react'
+import { getWeekdayNames } from '@/@types/Calendar'
+
+interface CalendarBodyDatesProps {
+  calendar: Calendar
+  getDateProps: (data: GetDatePropsOptions) => Record<string, any>
+}
+
+const CalendarBodyDates: FC<CalendarBodyDatesProps> = (props) => {
+  const { calendar, getDateProps } = props
+
+  const weekDayNames = getWeekdayNames()
+
+  const renderCalendarBody = () => {
+    return (
+      <>
+        {calendar.weeks.map((week, weekIndex) =>
+          week.map((dateObj, index) => {
+            const key = `${calendar.month}${calendar.year}${weekIndex}${index}`
+            if (!dateObj) {
+              return <WrapItem key={key} width={`calc(100% / 7)`}></WrapItem>
+            }
+            return renderCalendarDates(dateObj, key)
+          }),
+        )}
+      </>
+    )
+  }
```

Comment:
That's not what I meant. You should be able to create a new array from a `array.map` that returns react component and just use this array in rendering.